quantum/impl/quantum_task_impl.h

```diff
@@ -106,9 +108,39 @@ int Task::run()
         {
             return (int)ITask::RetCode::Sleeping;
         }
+        CoroutineState coroutineState = CoroutineState::Resumed;
+        if (_isNew)
+        {
+            coroutineState = CoroutineState::Constructed;
+            _isNew = false;
+        }
+        stateHandler(coroutineState);
```

Comment:
Depending on a coroutine, stateHandler may be called tons of times. Can we optimize this a bit by calling stateHandler only if stateHandler is defined?
i.e.
if (stateHandler) stateHandler(coroutineState);

Reply:
Sure